src/hooks/Auth.ts

```diff
@@ -294,11 +298,18 @@ export const useAuth = ({
         return;
       }
 
-      const extras = { ...options.endSessionRequest?.extras, ...endSessionRequest?.extras };
-
-      void performEndSessionRequest(endSessionHandler, configuration, options.clientId, options.redirectUrl, idToken, extras);
+      if (configuration.endSessionEndpoint) {
+        // Call endSession.
+        const extras = { ...options.endSessionRequest?.extras, ...endSessionRequest?.extras };
+        await performEndSessionRequest(endSessionHandler, configuration, options.clientId, options.redirectUrl, idToken, extras);
+        return true;
+      } else if (configuration.revocationEndpoint) {
+        // Only revoke the token.
+        await performRevokeTokenRequest(configuration, options.clientId, tmpIdToken);
```

Comment:
Token revocation refers to the refresh and access token, not the ID token. Cf. https://datatracker.ietf.org/doc/html/rfc7009#section-2

Reply:
Ok, so in the case of no endSessionEndpoint the only way to log out is to delete the token from the local store?

In that case I noticed that I still need to return the boolean which indicates if a manual redirect has to be done.
(Because no endSessionEndpoint can be called which then finally redirects)

An other idea would be to use the locationLike somehow to just directly redirect. That would introduce less special handling outside this lib...

Reply:
The ID token does not have to be revoked, as it is not an "access key", it is just a signed confirmation of the user's identity. 

For the other two, you should simply be able to call the revoke token endpoint with access and refresh token to perform the logout. 

Although technically, an RSA signed access token cannot be revoked as no communication with the IdP occurs on SP side upon validation. Removing it from the store should be enough.